app/src/test/kotlin/compiler/lexer/regex/RegexTest.kt

```diff
@@ -20,6 +20,102 @@ private val UNION_EP_EM = Regex.Union(EPSILON, EMPTY)
 private val UNION_EM_EP = Regex.Union(EMPTY, EPSILON)
 
 class RegexTest {
```

Comment:
Yeah, but to do it nicely I need to make RegexFactory an interface and move implementations to for example RegexFactoryImpl, so I can then change signature of derivative(a) to derivative(a, RegexFactory).
I was hoping I can perform this change after everything is merged.
With the current state I think I can still mock RegexFactory but it would be not possible to deduce from the test, where exactly is this used by subclasses of Regex.

Reply:
That's also why I skipped a few tests for derivative for now

Reply:
Ok sure

Reply:
Now, as I'm thinking, maybe it is too much effort :thinking: This code is pretty simple and tests would be probably just a repetition. So if you had some time to do it, then it is great, but if you don't, then it is also ok.